discord/poll.py

```diff
@@ -0,0 +1,641 @@
+        self.id: int = int(data['answer_id'])
+        self._message: Optional[Message] = message
+        self._state: Optional[ConnectionState] = message._state if message else None
+        self._vote_count: int = 0
+        self.self_voted: bool = False
+        self._poll: Poll = poll
+
+    def _handle_vote_event(self, added: bool) -> None:
+        if added:
+            self._vote_count += 1
+        else:
+            self._vote_count -= 1
+
+    def _update_with_results(self, payload: PollAnswerCountPayload) -> None:
+        self._vote_count = int(payload['count'])
+        self.self_voted = payload['me_voted']
+
+    def __str__(self) -> str:
+        return self.media.text
+
+    def __repr__(self) -> str:
+        return f'<PollAnswer id={self.id} media={self.media!r}>'
+
+    @classmethod
+    def from_params(
+        cls,
+        id: int,
+        text: str,
+        emoji: Optional[PollMediaEmoji] = None,
+        *,
+        poll: Poll,
+        message: Optional[Message],
+    ) -> Self:
+        poll_media: PollMediaPayload = {'text': text}
+        if emoji:
+            if isinstance(emoji, Emoji):
+                poll_media['emoji'] = {'name': emoji.name}  # type: ignore
+
+                if emoji.id:
+                    poll_media['emoji']['id'] = emoji.id  # type: ignore
+            elif isinstance(emoji, PartialEmoji):
+                poll_media['emoji'] = emoji.to_dict()
+            else:
+                poll_media['emoji'] = {'name': str(emoji)}  # type: ignore
+
+        payload: PollAnswerWithIDPayload = {'answer_id': id, 'poll_media': poll_media}
+
+        return cls(data=payload, message=message, poll=poll)
+
+    @property
+    def text(self) -> str:
+        """:class:`str`: Returns this answer display text."""
+        return self.media.text
+
+    @property
+    def emoji(self) -> Optional[Union[PartialEmoji, Emoji]]:
+        """Optional[:class:`PartialEmoji`]: Returns this answer display emoji, is any."""
+        if isinstance(self.media.emoji, str):
+            return PartialEmoji.from_str(self.media.emoji)
+        return self.media.emoji
+
+    @property
+    def vote_count(self) -> int:
+        """:class:`int`: Returns an approximate votes of this answer. If the poll
+        has finalized, it returns the exact amount of votes.
+        """
+        return self._vote_count
+
+    @property
+    def poll(self) -> Poll:
+        """:class:`Poll`: Returns the parent poll of this answer"""
+        return self._poll
+
+    def _to_dict(self) -> PollMediaPayload:
+        data: PollMediaPayload = {'text': self.text}
+
+        if self.emoji is not None:
+            if isinstance(self.emoji, PartialEmoji):
+                data['emoji'] = self.emoji.to_dict()
+            else:
+                data['emoji'] = {'name': str(self.emoji)}  # type: ignore
+
+                if hasattr(self.emoji, 'id'):
+                    data['emoji']['id'] = int(self.emoji.id)  # type: ignore
```

Comment:
```suggestion
    def _to_dict(self) -> PollAnswerPayload:
        return {
            'poll_media': self.media.to_dict(),
        }
```

Why isn't this returning the full poll answer payload?